app/scenes/Document/components/Document.tsx

```diff
@@ -138,6 +138,18 @@ class DocumentScene extends React.Component<Props> {
     }
   }
 
+  componentWillUnmount() {
+    if (
+      this.isEmpty &&
+      this.props.document.createdBy.id === this.props.auth.user?.id &&
+      this.props.document.isDraft &&
+      this.props.document.isActive &&
+      this.props.document.hasEmptyTitle
```

Comment:
a straightforward way to solve for the two people in the doc situation is to test for if there are any views on the doc that are not the author. I think that would be a very sane reason to retain the doc